### PR TITLE
#793 Href handles URI with fragment part

### PR DESCRIPTION
**Pull request by msarhan**

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#797](https://github.com/yegor256/takes/issues/797) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#797](https://github.com/yegor256/takes/issues/797) is assigned to @g4s8[/z](https://www.0crat.com/u/g4s8), here is [why](https://www.0crat.com/footprint/C3FFK3YAY/431733221). The budget is 15 minutes, see [§4](http://datum.zerocracy.com/pages/policy.html#4). Please, read [§27](http://datum.zerocracy.com/pages/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket. If you decide that this PR should not be accepted ever, also inform the architect.

File: src/test/java/org/takes/misc/HrefTest.java [hunk 1]
```diff
@@ -116,4 +116,20 @@ public void acceptsNonProperlyEncodedUrl() {
             Matchers.equalTo("http://www.netbout.com/%5Bfoo/bar%5D/read?file=%5B%5D%28%29.txt")
         );
     }
+
+    /**
+     * Href can build an URI with fragment.
+     */
+    @SuppressWarnings("PMD.AvoidDuplicateLiterals")
+    @Test
+    public void buildsUriWithFragment() {
```
Comment (g4s8) [inline src/test/java/org/takes/misc/HrefTest.java]: @msarhan there are two asserts inside one test method, please split them to two different methods (according to http://www.yegor256.com/2017/05/17/single-statement-unit-tests.html)

**Comment by g4s8** [comment]
rel #798

**Comment by msarhan** [comment]
@g4s8 unit test fixed.

**Comment by 0crat** [comment]
@g4s8[/z](https://www.0crat.com/u/g4s8) this job was assigned to you 5 days ago. It will be taken away from you soon, unless you close it, see [§8](http://datum.zerocracy.com/pages/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by g4s8** [comment]
@yegor256 ping

**Comment by 0crat** [comment]
@g4s8[/z](https://www.0crat.com/u/g4s8) this job was assigned to you 8 days ago. It will be taken away from you soon, unless you close it, see [§8](http://datum.zerocracy.com/pages/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by yegor256** [comment]
@rultor merge

**Comment by 0crat** [comment]
@elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova) please review this job, as in [§30](http://datum.zerocracy.com/pages/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by elenavolokhova** [comment]
@g4s8 According to our [QA rules](http://datum.zerocracy.com/pages/policy.html#42)
> The code reviewer found at least three problems in the code.

Only two issues were found during review.
Please, **confirm** that you'll try to find at least three design problems in next reviews and will pay attention to correct names and descriptions. 

Also, 
> Pull request description explains the solution proposed and contains a link to the original ticket it is related to.

This PR has empty description. As a code reviewer you should ask PR author to fix such issues too.

Thanks!

**Comment by elenavolokhova** [comment]
@msarhan Please, add details of implementation and the link to original issue to description.

**Comment by g4s8** [comment]
@elenavolokhova thanks for comments! I'll try to keep in mind all these rules next time.

**Comment by elenavolokhova** [comment]
@g4s8 Thanks!

**Comment by elenavolokhova** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
QA review completed: +15 points just awarded to @elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova), total is [+60](https://www.0crat.com/u/elenavolokhova)

**Pull request merged**